primeqa/mrc/processors/postprocessors/eli5_fid.py

```diff
@@ -0,0 +1,59 @@
+from primeqa.mrc.processors.postprocessors.abstract import AbstractPostProcessor
+from primeqa.mrc.data_models.eval_prediction_with_processing import EvalPredictionWithProcessing
+from transformers import PreTrainedTokenizerFast
+from datasets import Dataset
+from typing import List, Dict, Any, Tuple
+
+class ELI5FiDPostProcessor(AbstractPostProcessor):
+    """
+    Post processor for the generative FiD model on the ELI5 dataset.
+    """
+    def __init__(self,
+                 *args,
+                 tokenizer: PreTrainedTokenizerFast,
+                 **kwargs):
+        """
+        Args:
+            *args: Arguments for super class constructor.
+            **kwargs: Keyword Arguments for super class constructor.
+            tokenizer: Used to decode the output sequence generated by FiD
+        """
+        super().__init__(*args, **kwargs)
+        self.tokenizer = tokenizer
+        
+    def process(self, examples: Dataset, features: Dataset, predictions: tuple):
+         # Decode the predicted tokens.
+        preds = predictions.predictions
+        if isinstance(preds, tuple):
+            preds = preds[0]
+        decoded_preds = self.tokenizer.batch_decode(preds, skip_special_tokens=True)
+
+        # Build a map example to its corresponding features.
+        example_id_to_index = {k: i for i, k in enumerate(examples["id"])} 
+
+        feature_per_example = {example_id_to_index[feature["example_id"]]: i for i, feature in enumerate(features)}
```

Comment:
Does this work for the case that one example is mapped to multi features? or this never happens?